account_journal_general_sequence/models/account_journal.py

```diff
@@ -13,27 +13,20 @@ class AccountJournal(models.Model):
     entry_number_sequence_id = fields.Many2one(
         comodel_name="ir.sequence",
         string="Account entry number sequence",
-        default=lambda self: self._default_entry_number_sequence(),
+        compute="_compute_entry_number_sequence",
         copy=False,
         help="Sequence used for account entry numbering.",
     )
 
-    def _default_entry_number_sequence(self):
-        """Get the default sequence for all journals."""
-        result = self.env["ir.sequence"].search(
-            [("code", "=", "account_journal_general_sequence.default")]
-        )
-        if result:
-            return result
-        _logger.info("Creating default sequence for account move numbers")
-        result = self.env["ir.sequence"].create(
-            {
-                "name": _("Account entry default numbering"),
-                "code": "account_journal_general_sequence.default",
-                "implementation": "no_gap",
-                "prefix": "%(range_year)s/",
-                "padding": 10,
-                "use_date_range": True,
-            }
-        )
-        return result
+    def _compute_entry_number_sequence(self):
+        """Compute sequence for all journals."""
+        for rec in self:
+            sequence = False
+            if rec.company_id.name:
+                sequence = self.env["ir.sequence"].search(
+                    [
+                        ("code", "=", "account_journal_general_sequence.default"),
+                        ("company_id", "=", rec.company_id.id),
+                    ]
+                )
+            rec.entry_number_sequence_id = sequence.id
```

Comment:
Check if the chosen sequence is on another company instead of checking the name of the company and remove the default false value because we are going to do something if it's incorrect or it's not set.

Also, when assigning the sequence, can throw an exception if company name is falsy (can't happen for other constraint). Instead, assign the sequence only if the "if" happens (one more tab).

```suggestion
        is_model = self.env["ir.sequence"]
        for rec in self:
            if rec.entry_number_sequence_id.company_id != rec.company_id:
                rec.entry_number_sequence_id = is_model.search(
                    [
                        ("code", "=", "account_journal_general_sequence.default"),
                        ("company_id", "=", rec.company_id.id),
                    ], limit=1
                ).id
```

Reply:
According to my other comments, this will not be necessary because we shouldn't be using a compute after all.